src/NodaTime/Utility/Preconditions.cs

```diff
@@ -112,11 +112,13 @@ internal static void DebugCheckArgumentRange([InvokerParameterName] string param
         [Conditional("DEBUG")]
         internal static void DebugCheckArgument(bool expression, [InvokerParameterName] string parameter, string messageFormat, params object[] messageArgs)
         {
+#if DEBUG
```

Comment:
I needed to introduce DebugCheckState, and that led me to find the inconsistent exception use. It may well be worth me pulling out these changes into a separate PR though...